### PR TITLE
docs: generate Typedocs using Aegir

### PR DESCRIPTION
**Pull request by maschad**
Closes https://github.com/libp2p/js-libp2p/issues/1455

### CODE REVIEW

**Comment by maschad** [comment]
Once https://github.com/ipfs/aegir/pull/1107 is merged and a new version of `aegir` is released then this can be reviewed.

File: README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ If you are looking for the documentation of the latest release, you can view the
   - [Tests](#tests)
     - [Run unit tests](#run-unit-tests)
   - [Packages](#packages)
+- [Used by](#used-by)
```
Comment (maschad) [inline README.md:55]: Good catch! thanks

**Comment by maschad** [comment]
Not sure why CI was failing, rebased and pushed again.

**Comment by maschad** [comment]
@achingbrain quite a few of the parameters are not exported as modules and therefore no documentation is generated, **but** they are still referenced by other generated documentation e.g. `ConnectionManager` or `IdentifyService`  - they are also included in the [current API Docs](https://github.com/libp2p/js-libp2p/blob/master/doc/API.md), so do we want to include these?  They may not be modules that we want to export but I don't see the downside of exposing them.

**Comment by achingbrain** [comment]
> quite a few of the parameters are not exported as modules and therefore no documentation is generated, but they are still referenced by other generated documentation 

I haven't tried it yet but this might help? https://www.npmjs.com/package/typedoc-plugin-missing-exports

**Comment by maschad** [comment]
Thanks for that recommendation @achingbrain , once https://github.com/ipfs/aegir/pull/1131 is merged and released then I will re-open this for review.

**Review by achingbrain — APPROVED** [review]

**Pull request merged**